ClosedXML/Excel/Style/IXLStylized.cs

```diff
@@ -6,11 +6,18 @@ namespace ClosedXML.Excel
     internal interface IXLStylized
     {
         IXLStyle Style { get; set; }
+
         IEnumerable<IXLStyle> Styles { get; }
-        Boolean UpdatingStyle { get; set; }
+
         IXLStyle InnerStyle { get; set; }
+
         IXLRanges RangesUsed { get; }
-        Boolean StyleChanged { get; set; }
-        //Boolean IsDefault { get; set; }
+
+        /// <summary>
+        /// Immutable style
+        /// </summary>
+        XLStyleValue StyleValue { get; }
```

Comment:
Can we not keep all the new value and key classes/structs out of the API and internal to ClosedXML? Surely the user should not be concerned wit the internal implementation?

Reply:
This will require performing a typecast each time we want to get a style value from ```IXLSomething``` (roughly, 50 places in code) but this is surely possible.

Reply:
From the `Sandbox` project, the `XLFillKey` and `XLFillValue` classes are visible, because they are public. I would expect them to be internal and that the API still expose only the `IXLFill` interface. Same for the other style classes. Where would the typecast be needed?